src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -161,6 +240,430 @@ public void ContinueTimeout_SetNegativeTwo_ThrowsArgumentOutOfRangeException(Uri
+            request.PreAuthenticate = false;
+            Assert.False(request.PreAuthenticate);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Connection_NullOrWhiteSpace_ValuesMatch(Uri remoteServer)
+        {
+            const string connection = "connect";
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.Connection = connection;
+            Assert.Equal(connection, request.Connection);
+            request.Connection = null;
+            Assert.Equal(null, request.Connection);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Connection_SetKeepAliveAndClose_ThrowsArgumentException(Uri remoteServer)
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            Assert.Throws<ArgumentException>(() => request.Connection = "keep-alive");
+            Assert.Throws<ArgumentException>(() => request.Connection = "close");
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Expect_SetNullOrWhiteSpace_ValuesMatch(Uri remoteServer)
+        {
+            const string expect = "101-go";
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.Expect = expect;
+            Assert.Equal(expect, request.Expect);
+            request.Expect = null;
+            Assert.Equal(null, request.Expect);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Expect_Set100Continue_ThrowsArgumentException(Uri remoteServer)
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            Assert.Throws<ArgumentException>(() => request.Expect = "100-continue");
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void DefaultMaximumResponseHeadersLength_SetAndGetLength_ValuesMatch(Uri remoteServer)
+        {
+            // Save default value
+            int defaultMaximumResponseHeadersLength = HttpWebRequest.DefaultMaximumResponseHeadersLength;
+
+            int newDefaultMaximumResponseHeadersLength = 255;
+            HttpWebRequest.DefaultMaximumResponseHeadersLength = newDefaultMaximumResponseHeadersLength;
+            Assert.Equal(newDefaultMaximumResponseHeadersLength, HttpWebRequest.DefaultMaximumResponseHeadersLength);
+
+            // Cleanup
+            HttpWebRequest.DefaultMaximumResponseHeadersLength = defaultMaximumResponseHeadersLength;
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void DefaultMaximumErrorResponseLength_SetAndGetLength_ValuesMatch(Uri remoteServer)
+        {
+            // Save default value
+            int defaultMaximumErrorsResponseLength = HttpWebRequest.DefaultMaximumErrorResponseLength;
+
+            int newDefaultMaximumErrorsResponseLength = 255;
+            HttpWebRequest.DefaultMaximumErrorResponseLength = newDefaultMaximumErrorsResponseLength;
```

Comment:
Great, thanks for the info! Still I think we should be safe as we aren't touching any parts of HttpWebRequest in other tests classes.